mono/mini/debugger-agent.c

```diff
@@ -1861,7 +1861,8 @@ stop_debugger_thread (void)
 			mono_coop_mutex_unlock (&debugger_thread_exited_mutex);
 		} while (!debugger_thread_exited);
 
-		mono_native_thread_join (debugger_thread_id);
```

Comment:
@joncham This removes your previous fix. As far as I can tell this is solving the same problem in a different way.

Reply:
Yes, I think it's equivalent.